src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs

```diff
@@ -26,17 +87,507 @@ public override bool CanHandleType(SqlObjectType objectType)
+            if (!IsAnyManagedInstance(dataContainer.Server))
+            {
+
+                // add recovery model options to the dropdown
+                recoveryModels.Add(displayRecoveryModels[RecoveryModel.Full]);
+                recoveryModels.Add(displayRecoveryModels[RecoveryModel.BulkLogged]);
+                recoveryModels.Add(displayRecoveryModels[RecoveryModel.Simple]);
+            }
+            else
+            {
+                if (prototype.OriginalName.Equals("tempdb", StringComparison.CurrentCultureIgnoreCase) && prototype.IsSystemDB)
+                {
+                    // tempdb supports 'simple recovery' only
+                    recoveryModels.Add(displayRecoveryModels[RecoveryModel.Simple]);
+                }
+                else
+                {
+                    // non-tempdb supports only 'full recovery' model
+                    recoveryModels.Add(displayRecoveryModels[RecoveryModel.Full]);
+                }
+            }
+
+            if (recoveryModelEnabled)
+            {
+                var swapIndex = 0;
+                switch (prototype.RecoveryModel)
+                {
+                    case RecoveryModel.BulkLogged:
+                        swapIndex = 1;
+                        break;
+
+                    case RecoveryModel.Simple:
+                        swapIndex = 2;
+                        break;
+
+                    default:
+                        System.Diagnostics.Debug.Assert(RecoveryModel.Full == prototype.RecoveryModel, "unexpected recovery model");
+                        break;
+                }
+                if (swapIndex > 0)
+                {
+                    var value = recoveryModels[swapIndex];
+                    recoveryModels.RemoveAt(swapIndex);
+                    recoveryModels.Insert(0, value);
+                }
+            }
+            return recoveryModels.ToArray();
+        }
+
+        private string[] PopulateCompatibilityLevelDropdownAzure(CDataContainer dataContainer, DatabasePrototype prototype)
+        {
+            // For Azure we loop through all of the possible compatibility levels. We do this because there's only one compat level active on a
+            // version at a time, but that can change at any point so in order to reduce maintenance required when that happens we'll just find
+            // the one that matches the current set level and display that
+            foreach (var level in this.displayCompatLevels.Keys)
+            {
+                if (level == prototype.DatabaseCompatibilityLevel)
+                {
+                    // Azure can't change the compat level so we only populate the current version
+                    return new string[] { this.displayCompatLevels[level] };
+                }
+            }
+
+            System.Diagnostics.Debug.Fail(string.Format(CultureInfo.InvariantCulture, "Unknown compat version '{0}'", prototype.DatabaseCompatibilityLevel));
```

Comment:
This seems like it should be just a throw, not sure having Debug.Fail calls is actually useful (it likely just means that if there's ever an issue during release it'll just fail silently or elsewhere)

Reply:
A bunch of these methods were ported over from SSMS, so I was trying to maintain the same behavior.

Reply:
That's fine to start with, but SSMS and STS are very different applications - so we should be doing what makes sense for STS where we can. Debug.Fail is only marginally useful even normally (especially in a situation like this where it's clearly indicating a runtime error has occurred and there isn't really a way to recover from it) so seems like an easy thing to start with replacing.